src/bleak_retry_connector/__init__.py

```diff
@@ -427,22 +427,27 @@ def _raise_if_needed(name: str, description: str, exc: Exception) -> None:
     raise RuntimeError("This should never happen")
 
 
-WrapFuncType = TypeVar("WrapFuncType", bound=Callable[..., Any])
+P = ParamSpec("P")
+T = TypeVar("T")
 
 
-def retry_bluetooth_connection_error(attempts: int = DEFAULT_ATTEMPTS) -> WrapFuncType:  # type: ignore[type-var]
+def retry_bluetooth_connection_error(
+    attempts: int = DEFAULT_ATTEMPTS,
+) -> Callable[[Callable[P, Awaitable[T]]], Callable[P, Awaitable[T]]]:
     """Define a wrapper to retry on bluetooth connection error."""
 
-    def _decorator_retry_bluetooth_connection_error(func: WrapFuncType) -> WrapFuncType:
+    def _decorator_retry_bluetooth_connection_error(
+        func: Callable[P, Awaitable[T]]
+    ) -> Callable[P, Awaitable[T]]:
         """Define a wrapper to retry on bleak error.
 
         The accessory is allowed to disconnect us any time so
         we need to retry the operation.
         """
 
-        async def _async_wrap_bluetooth_connection_error_retry(
-            *args: Any, **kwargs: Any
-        ) -> Any:
+        async def _async_wrap_bluetooth_connection_error_retry(  # type: ignore[return]
```

Comment:
The mypy ignore here is needed because mypy expects the function to also return outside of the `for` loop. But there isn't anything sensible we could return there.